src/jenkins/BuildManifest.groovy

```diff
@@ -25,30 +25,76 @@ class BuildManifest implements Serializable {
         String getFilename() {
             return this.name.toLowerCase().replaceAll(' ', '-')
         }
+
+        String getPackageName() {
+            return [
+                    this.getFilename(),
+                    this.version,
+                    this.platform,
+                    this.architecture,
+            ].join('-') + '.tar.gz'
+        }
+    }
+
+    class Components extends HashMap<String, Component> {
+
+        Components(ArrayList data) {
+            data.each { item ->
+                Component component = new Component(item)
+                this[component.name] = component
+            }
+        }
+
+        public String getMinArtifact(){
+            try {
+                this.get("OpenSearch").artifacts.get("dist").first()
+            } catch (Exception ex){
+                echo("Min artifact cannot be found", ex.getMessage())
```

Comment:
Name is the only thing that will distinguish them. However, I checked their build manifest and the build.name is a way to differentiate between opensearch and dashboards. How about this:

```
        public String getMinArtifact(){
            if (build.name == "OpenSearch") {
                this.get("OpenSearch")?.artifacts?.get("dist")?.first()
            } else if (build.name == "OpenSearch Dashboards"){
                this.get("OpenSearch-Dashboards")?.artifacts?.get("dist")?.first()
            }
        }
```

Reply:
Since it's a function it should just either return `null` or raise an error.

I would just do `this.get(build.name.replace(' ', '-'))?....` and call it a day.

Reply:
done! Thanks!